homeassistant/components/reisinger_intellidrive/cover.py

```diff
@@ -0,0 +1,136 @@
+"""The cover entity of reisinger intellidrive."""
+
+import logging
+from typing import Any, cast
+
+from homeassistant.components.cover import (
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_TOKEN,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, STATUSDICT_OPENSTATE, STATUSDICT_SERIALNO
+from .coordinator import IntellidriveCoordinator
+from .entity import IntelliDriveEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+STATES_MAP = {0: STATE_CLOSED, 1: STATE_OPEN}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Intellidrive covers."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            SlidingDoorCoverEntity(
+                hass,
+                coordinator,
+                entry.data[CONF_HOST],
+                str(entry.data.get(CONF_TOKEN)),
+                cast(str, entry.unique_id),
+            )
+        ]
+    )
+
+
+class SlidingDoorCoverEntity(IntelliDriveEntity, CoverEntity):
+    """Wrapper class to adapt the intellidrive device into the Homeassistant platform."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        coordinator: IntellidriveCoordinator,
+        host: str,
+        token: str,
+        device_id: str,
+    ) -> None:
+        """Initialize slidingdoor cover."""
+        self._state: str | None = None
+        self._state_before_move: str | None = None
+        self._host = host
+        self._token = token
+        self._device_api = coordinator.device
+        status = coordinator.data
+        self._attr_name = f"Slidingdoor {status[STATUSDICT_SERIALNO]}"
```

Comment:
For new integrations, we require the following:
- Support for [the new entity naming](https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations)
- Services must be [translatable](https://developers.home-assistant.io/blog/2023/07/11/translating-services)
- Entities and their attributes should be [translatable](https://developers.home-assistant.io/blog/2023/03/27/entity_name_translations/)

You can't set anymore a name like this. Set it to `None` if the entity should use the device name or use the translation key.